mythril/src/apic.rs

```diff
@@ -103,6 +105,32 @@ pub unsafe fn get_local_apic_mut() -> &'static mut LocalApic {
         .expect("Attempt to get local APIC before initialization")
 }
 
+/// A representation of a APIC ID
+#[derive(Copy, Clone, Debug, Ord, PartialEq, PartialOrd, Eq)]
+pub struct ApicId {
+    /// The raw ID as an integer
+    pub raw: u32,
+}
+
+impl ApicId {
+    /// Returns whether this is the BSP core
```

Comment:
We could also get this from the `IA32_APIC_BASE_MSR`... Not sure if there is a reason to do that instead of this, but it might be more robust

Reply:
I don't think we can really use that, because it would just tell us if the current APIC is a BSP, but we need to check if a given id is the id of a BSP, which we could only really do by recording that info somewhere and looking it up. I added a note that the current logic is incorrect for multi socket systems.

Reply:
:+1: that makes sense